src/quadrature/quadrature.cc

```diff
@@ -0,0 +1,128 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * 
+ * @file    quadrature.hh
+ * @brief   Basic definitions of numerical quadrature rules.
+ * @author  Jan Stebel
+ */
+
+#include "quadrature/quadrature.hh"
+
+
+
+Quadrature& Quadrature::operator=(const Quadrature &q)
+{
+    ASSERT_DBG( dim_ == q.dim_ );
+    quadrature_points = q.quadrature_points;
+    weights = q.weights;
+    return *this;
+}
+    
+
+Quadrature::Quadrature(unsigned int dimension, const unsigned int n_q)
+: dim_(dimension),
+  quadrature_points(n_q, dimension),
+  weights(n_q, 0)
+{}
+
+
+Quadrature::Quadrature(const Quadrature &q) :
+        dim_(q.dim_),
+        quadrature_points(q.quadrature_points),
+        weights(q.weights)
+{}
+
+
+
+
+
+
+
+template<unsigned int quad_dim>
+Quadrature quadrature_from_side(const Quadrature &subq, unsigned int sid, unsigned int pid)
```

Comment:
I suppose this is a replacement for the constructor? However as the diff takes it as a completely new code it is hard to see the changes. Better discuss it personally.

Reply:
Yes, this method was originally constructor. I was not able to declare a templated constructor, hence moved it to a templated function.